File: 0467-Unique Substrings in Wraparound String/0467-Unique Substrings in Wraparound String.cpp
class Solution {
public:
    int findSubstringInWraproundString(string p) {
        vector<int> count(26);
        int maxLen = 1;
        for (int i = 0; i < p.size(); ++i)
        {
            if (i > 0 && (p[i - 1] - 'a' == (p[i] - 'a' + 25) % 26))
            {
                ++maxLen;
            }
            else
            {
                maxLen = 1;
            }
            
            count[p[i] - 'a'] = max(count[p[i] - 'a'], maxLen);
        }
        
        return accumulate(count.begin(), count.end(), 0);
    }
};

File: 0401-0500/0467-Unique Substrings in Wraparound String/0467-Unique Substrings in Wraparound String.py
class Solution:
    def findSubstringInWraproundString(self, p: 'str') -> 'int':
        count = [0]*26
        maxLen = 1
        for i in range(len(p)):
            if i > 0 and (ord(p[i]) - ord(p[i - 1]))%26 == 1:
                maxLen += 1
            else:
                maxLen = 1
            count[ord(p[i]) - ord('a')] = max(count[ord(p[i]) - ord('a')], maxLen)
        return sum(count)
